src/runtime/inline-cta-api.ts

```diff
@@ -74,15 +70,76 @@ export class InlincCtaApi {
     return feUrl(urlPrefix, iframeParams);
   }
 
+  private handleCompleteAudienceActionResponse_(
+    response: CompleteAudienceActionResponse,
+    actionType: string,
+    div: HTMLElement
+  ): void {
+    this.entitlementsManager_.clear();
+    const userToken = response.getSwgUserToken();
+    if (userToken) {
+      this.storage_.set(StorageKeys.USER_TOKEN, userToken, true);
+    }
+    if (response.getAlreadyCompleted()) {
+      this.clearInlineCta_(div);
+      this.showAlreadyOptedInToast_(actionType);
+    }
+    const now = Date.now().toString();
+    this.storage_.set(StorageKeys.READ_TIME, now, /*useLocalStorage=*/ false);
+    this.entitlementsManager_.getEntitlements();
+  }
+
+  private showAlreadyOptedInToast_(actionType: string): void {
```

Comment:
Does it make sense to refactor the original function to make it reusable here?

Reply:
Created a shared util for the function.